### PR TITLE
Serde for Iterators, Added new messages

### PR DESCRIPTION
**Pull request by adrian-kong**
This PR adds some of missing messages (before recent merges) and code-gen for serde serialize implementations, helpful for serializing iterators.

List of new messages:
- ESF_MEAS,
- RXM_RAWX
- RXM_SFRBX
- ESF_RAW
- ESF_INS
- HNR_PVT
- NAV_ATT
- NAV_CLOCK
- NAV_VELECEF
- MGA_GPA_EPH

Serialization outputs includes the class id and message id*

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     - name: Install MSRV
       uses: actions-rs/toolchain@v1
       with:
-        toolchain: 1.57.0
+        toolchain: 1.63.0
```
Comment (lkolbly) [inline .github/workflows/rust.yml:24]: Hey! Sorry for the delayed review.

Do we need 1.63 for this change, or can we get by with 1.57? I'm trying to keep the MSRV around a year old or so.

Comment (adrian-kong) [inline .github/workflows/rust.yml:24]:
```suggestion
        toolchain: 1.57.0
```

Comment (adrian-kong) [reply .github/workflows/rust.yml]: I don't think so, can try reverting this. Besides that, is everything all good?

File: ublox_cli/src/main.rs [hunk 2]
```diff
@@ -1,196 +1,20 @@
...
-                if let PacketRef::AckAck(ack) = packet {
-                    if ack.class() == T::CLASS && ack.msg_id() == T::ID {
-                        found_packet = true;
-                    }
-                }
-            })?;
-        }
-        Ok(())
-    }
+use std::io;
```
Comment (lkolbly) [inline ublox_cli/src/main.rs:1]: Would it make sense to move this to a different example crate, or add it as another option alongside the previous code?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (lkolbly) [reply .github/workflows/rust.yml]: I just commented about the ublox_cli changes, but then yeah everything looks good. (it looks like there might be some merge conflicts, I haven't gotten a chance to look at them yet)

File: ublox_cli/src/main.rs (hunk 2 above)
Comment (adrian-kong) [reply ublox_cli/src/main.rs]: Yeah was just testing for our use case, this has been reverted though. Passes CI as well with reverted MSRV

File: ublox/Cargo.toml [hunk 3]
```diff
@@ -9,7 +9,7 @@ repository = "https://github.com/lkolbly/ublox"
 readme = "../README.md"
 
 [features]
-default = ["std"]
+default = ["std", "serde"]
```
Comment (silverjam) [inline ublox/Cargo.toml:12]: Should we disable serde by default?

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -17,7 +17,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Install libudev
-      run: sudo apt-get install -y libudev-dev
+      run: sudo apt-get update && sudo apt-get install -y libudev-dev
```
Comment (silverjam) [inline .github/workflows/rust.yml:20]: We need a build variant that ensures that the library can build with `serde` enabled, correct?

**Review by silverjam — COMMENTED** [review]
@lkolbly Can you please enable CI checks for this PR?

**Comment by lkolbly** [comment]
@silverjam I think CI is currently enabled for this PR?

(also, I should be able to merge this this weekend, hopefully)

**Comment by lkolbly** [comment]
There are some merge conflicts though, maybe that's blocking CI.

File: ublox/Cargo.toml (hunk 3 above)
Comment (lkolbly) [reply ublox/Cargo.toml:12]: I think serde by default should be fine.

File: .github/workflows/rust.yml (hunk 4 above)
Comment (lkolbly) [reply .github/workflows/rust.yml]: We also need a matrix that tests we can build no_std and no alloc, for now I think we can survive with just testing the default case and we'll add that later.

**Comment by lkolbly** [comment]
@silverjam Sorry for the delay, yeah apparently it wasn't enabled, not sure what was up with that. Anyway I enabled CI and it looks like we just need to `use serde::{Serialize, Deserialize}`

**Comment by silverjam** [comment]
@lkolbly Ok, builds and tests are passing locally for me -- looks like you'll need to approve the CI one more time

**Comment by silverjam** [comment]
Thanks for getting this in @lkolbly!

**Pull request merged**